AIX build failure due to incompatible struct types

While building logrotate in AIX, we are facing build issues, below is the log

```

gcc -DHAVE_CONFIG_H -I.  -include config.h  -Wall -Wconversion -Wdeclaration-after-statement -Wextra -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -g -O2 -MT logrotate.o -MD -MP -MF .deps/logrotate.Tpo -c -o logrotate.o logrotate.c
logrotate.c: In function 'setAtimeMtime':
logrotate.c:805:13: error: incompatible types when assigning to type 'struct timespec' from type 'st_timespec_t' {aka 'const struct st_timespec'}
  805 |     ts[0] = sb->st_atim;
      |             ^~
logrotate.c:806:13: error: incompatible types when assigning to type 'struct timespec' from type 'st_timespec_t' {aka 'const struct st_timespec'}
  806 |     ts[1] = sb->st_mtim;
  logrotate.c: At top level:
logrotate.c:1155: warning: "MIN" redefined
 1155 | #define MIN(a,b) ((a) < (b) ? (a) : (b))
      | 
In file included from /usr/include/sys/file.h:29,
                 from logrotate.c:11:
/opt/freeware/lib/gcc/powerpc-ibm-aix7.1.0.0/10/include-fixed/sys/param.h:183: note: this is the location of the previous definition
  183 | #define MIN(a,b) (((a)<(b))?(a):(b))
      | 
make[2]: *** [Makefile:531: logrotate.o] Error 1
make[2]: Leaving directory '/home/buildusr/rpmbuild/BUILD/logrotate-3.21.0/32bit
```

--- a/logrotate.c
+++ b/logrotate.c
@@ -1194,8 +1194,6 @@ static int is_probably_sparse(struct stat const *sb)
 #endif
 }
 
-#define MIN(a,b) ((a) < (b) ? (a) : (b))
-
 /* Return whether the buffer consists entirely of NULs.
    Note the word after the buffer must be non NUL. */
 
@@ -1251,7 +1249,8 @@ static int sparse_copy(int src_fd, int dest_fd, const struct stat *sb,
     while (max_n_read) {
         int make_hole = 0;
         size_t bytes_read;
-        const ssize_t n_read = read (src_fd, buf, MIN (max_n_read, BUFSIZ));
+        const size_t to_read = (max_n_read < BUFSIZ) ? max_n_read : BUFSIZ;
+        const ssize_t n_read = read (src_fd, buf, to_read);
         if (n_read < 0) {
             if (errno == EINTR) {
                 continue;

--- a/logrotate.c
+++ b/logrotate.c
@@ -841,16 +841,24 @@ static void setAtimeMtime(int fd, const char *filename, const struct stat *sb)
 #if defined HAVE_FUTIMENS && defined HAVE_STRUCT_STAT_ST_ATIM && defined HAVE_STRUCT_STAT_ST_MTIM
     struct timespec ts[2];
 
-    ts[0] = sb->st_atim;
-    ts[1] = sb->st_mtim;
+    /* Copy field-by-field: on some platforms (e.g. AIX) the stat
+       timestamp members are a distinct struct type from struct timespec. */
+    ts[0].tv_sec = sb->st_atim.tv_sec;
+    ts[0].tv_nsec = sb->st_atim.tv_nsec;
+    ts[1].tv_sec = sb->st_mtim.tv_sec;
+    ts[1].tv_nsec = sb->st_mtim.tv_nsec;
     futimens(fd, ts);
 
     (void)filename;
 #elif defined HAVE_UTIMENSAT && defined HAVE_STRUCT_STAT_ST_ATIM && defined HAVE_STRUCT_STAT_ST_MTIM
     struct timespec ts[2];
 
-    ts[0] = sb->st_atim;
-    ts[1] = sb->st_mtim;
+    /* Copy field-by-field: on some platforms (e.g. AIX) the stat
+       timestamp members are a distinct struct type from struct timespec. */
+    ts[0].tv_sec = sb->st_atim.tv_sec;
+    ts[0].tv_nsec = sb->st_atim.tv_nsec;
+    ts[1].tv_sec = sb->st_mtim.tv_sec;
+    ts[1].tv_nsec = sb->st_mtim.tv_nsec;
     utimensat(AT_FDCWD, filename, ts, AT_SYMLINK_NOFOLLOW);
 
     (void)fd;

